handlers/handlers.go

```diff
@@ -395,121 +395,134 @@ func readPreviousReleases(w http.ResponseWriter, req *http.Request, cfg *config.
 ) {
 	ctx, cancel := context.WithCancel(req.Context())
 	defer cancel()
-	var err error
 	template := "related-list-pages"
 	urlPath := path.Dir(req.URL.Path)
 	urlQuery := req.URL.Query()
 	latestContentURL := urlPath + "/latest"
 
-	sanitisedParams := sanitiseQueryParams(allowedPreviousReleasesQueryParams, urlQuery)
-	// check page type
-	pageData, err := zc.GetPageData(ctx, accessToken, collectionID, lang, latestContentURL)
-	if err != nil {
-		log.Error(ctx, "failed to get content type", err)
-		w.WriteHeader(http.StatusNotFound)
-		return
-	}
-	if !slices.Contains(knownPreviousReleaseTypes, pageData.Type) {
-		err := errors.New("page type doesn't match known list of content types compatible with /previousreleases")
-		log.Error(ctx, "page type isn't compatible with /previousreleases", err)
-		w.WriteHeader(http.StatusNotFound)
-		return
-	}
-	// get cached navigation data
-	navigationCache, err := cacheList.Navigation.GetNavigationData(ctx, lang)
-	if err != nil {
-		log.Error(ctx, "failed to get navigation cache for aggregation", err)
-		setStatusCode(w, req, err)
-		return
-	}
-
-	validatedQueryParams, validationErrs := data.ReviewPreviousReleasesQueryWithParams(ctx, cfg, sanitisedParams, urlPath)
-
-	if len(validationErrs) > 0 {
-		log.Info(ctx, "validation of parameters failed for aggregation", log.Data{
-			"parameters": validationErrs,
-		})
-		// Errors are now mapped to the page model to output feedback to the user rather than
-		// a blank 400 error response.
-		m := mapper.CreatePreviousReleasesPage(cfg, req, rend.NewBasePageModel(), validatedQueryParams, &searchModels.SearchResponse{}, lang, zebedeeCli.HomepageContent{}, "", navigationCache, template, cache.Topic{}, validationErrs, zebedeeCli.PageData{}, []zebedeeCli.Breadcrumb{})
-		buildDataAggregationPage(w, m, rend, template)
-		return
-	}
-
 	// counter used to keep track of the number of concurrent API calls
-	var counter = 3
-	searchQuery := data.SetParentTypeOnSearchAPIQuery(validatedQueryParams, pageData.Type)
-
+	var counter = 4
 	var (
-		homepageResp zebedeeCli.HomepageContent
-		searchResp   = &searchModels.SearchResponse{}
-		bc           []zebedeeCli.Breadcrumb
+		pageData        zebedeeCli.PageData
+		navigationCache *models.Navigation
+		bc              []zebedeeCli.Breadcrumb
+		homepageResp    zebedeeCli.HomepageContent
 
 		wg sync.WaitGroup
 
-		respErr, countErr, bcErr error
+		pdErr, ptErr, nErr, bcErr, homeErr error
 	)
+
 	wg.Add(counter)
 
 	go func() {
 		defer wg.Done()
-		var homeErr error
-		homepageResp, homeErr = zc.GetHomepageContent(ctx, accessToken, collectionID, lang, homepagePath)
-		if homeErr != nil {
-			log.Warn(ctx, "unable to get homepage content", log.FormatErrors([]error{homeErr}))
+		// check page type
+		pageData, pdErr = zc.GetPageData(ctx, accessToken, collectionID, lang, latestContentURL)
+		if pdErr != nil {
+			log.Error(ctx, "failed to get content type", pdErr)
+			cancel()
+			return
+		}
```

Comment:
The reason this was structured non-concurrently was to prevent additional calls if we shouldn't render the page at all.

Reply:
We were never handling the error, just logging it. This was the page still renders